docs/guides/table_migration.md

```diff
@@ -0,0 +1,120 @@
+# Table migration
+
+SQLMesh projects can read directly from tables not managed by SQLMesh, but in some cases it may be desirable to migrate an existing table into a SQLMesh project. This guide describes two methods for migrating existing tables into a SQLMesh project.
+
+We assume that downstream consumers will continue to select from an object with the original table name, so the migration task is to convert an existing table into an identically named SQLMesh model. We also assume that the table is incremental, as there would be no need for migration if the table could be rebuilt on every run.
+
+## Do you need to migrate?
+
+SQLMesh does not assume it manages all data sources: models can read from any data source accessible by the SQL engine, treating them as as [external models](../concepts/models/model_kinds.md#external) that include column-level lineage or as generic sources. This approach is preferred to migrating existing tables into a SQLMesh project.
+
+You should only migrate tables if the source table you are working with is either too large to be rebuilt or cannot be rebuilt because the necessary historical data is unavailable.
```

Comment:
Let's also highlight that this is only relevant to tables that are already being populated incrementally.